src/components/basic/DlToggleButton/DlToggleButton.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+    <div
+        :style="{ width }"
+        class="container"
+    >
+        <dl-button
+            v-for="(btn, idx) in toggleButtons"
+            :key="idx"
+            :styles="
+                getStyles(btn.value === scopedValue, hoverBtn === btn.value)
+            "
+            data-test="button"
+            fluid
+            @mouseenter="hoverBtn = btn.value"
+            @mouseleave="hoverBtn = null"
+            @click="value = btn.value"
+        >
+            <span v-if="!$slots.button && !$slots[`button-${idx}`]">
+                {{ btn.label }}
+            </span>
+            <slot
+                :label="btn.label"
+                :name="`button-${idx}`"
+            />
+            <slot
+                :label="btn.label"
+                name="button"
+            />
+        </dl-button>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import { OptionItem } from './types'
```

Comment:
This type should be exported to everyone please c hange the name to DlToggleButtonOption
and make sure to export it in the types